docs/extend/backend-events.md

```diff
@@ -0,0 +1,125 @@
+# Backend Events
+
+Often, an extension will want to react to some events occuring elsewhere in Flarum. For instance, we might want to increment a counter when a new discussion is posted, send a welcome email when a user logs in for the first time, or add tags to a discussion before saving it to the database. These events are known as **domain events**, and are broadcast across the framework through [Laravel's event system](https://laravel.com/docs/6.x/events).
+
+:::warning Old Event API
+Historically, Flarum has used events for its extension API, emitting events like `GetDisplayName` or `ConfigureApiRoutes` to allow extensions to insert logic into various parts of Flarum. These events are gradually being phased out in favor of the declarative [extender system](start.md#extenders), and will be removed before stable. Domain events will not be removed.
+:::
+
+For a full list of backend events, see our [API documentation](https://api.docs.flarum.org/php/master/search.html?search=Event). Domain events classes are organized by namespace, usually `Flarum\TYPE\Event`.
+
+## Listening to Events
+
+You can attach a listener to an event using the [`Event`](https://api.docs.flarum.org/php/master/flarum/extend/event) [extender](start.md#extenders):
+
+```php
+use Flarum\Extend;
+use Flarum\Post\Event\Deleted;
+use Symfony\Component\Translation\TranslatorInterface;
+
+
+return [
+    (new Extend\Event)
+        ->listen(Deleted::class, function($event) {
+          // do something here
+        })
+        ->listen(Deleted::class, PostDeletedListener::class)
+];
+
+
+class PostDeletedListener
+{
+  protected $translator;
+
+  public function __construct(TranslatorInterface $translator)
+  {
+      $this->translator = $translator;
+  }
+
+  public function handle(Deleted $event)
+  {
+    // Your logic here
+  }
+}
+```
+
+As shown above, a listener class can be used instead of a callback. This allows you to [inject dependencies](https://laravel.com/docs/6.x/container) into your listener class.
```

Comment:
```suggestion
As shown above, a listener class can be used instead of a callback. This allows you to [inject dependencies](https://laravel.com/docs/6.x/container) into your listener class via constructor parameters. In this example we resolve a translator instance.
```

I feel like people might not understand having the `TranslatorInterface` in the constructor is **optional**.